webapp/src/components/modals/create_issue/create_issue_form.tsx

```diff
@@ -223,7 +223,7 @@ export default class CreateIssueForm extends React.PureComponent<Props, State> {
 
         const result = {} as CreateIssueFields;
         for (const key of Object.keys(fields)) {
-            if (available[key] || key === 'project' || key === 'issuetype') {
+            if ((available[key] || key === 'project' || key === 'issuetype') && key !== 'assignee') {
```

Comment:
@mickmister tested this a couple of times and this fixes the issue but for me this is more of a hot fix as the underlying issue is that the types don't take into account the assignee and the user selector component which seems to have conflicts with the type as well. Maybe we can create a seperate issue that fixes it.

Reply:
There are other custom user selector fields besides `assignee`, so this fix won't fix it for all cases of having a user selector.

Something to keep in mind is that different projects and different issue types can have entirely different field schemas.

There is a block in `create_issue_form.tsx` that copies all the previous fields to the new form state:

https://github.com/mattermost/mattermost-plugin-jira/blob/538f68a79144890463476474b10b33893bfa41df/webapp/src/components/modals/create_issue/create_issue_form.tsx#L138-L141

I think we should make it so the `summary` and `description` are the only fields that we keep around, in the `handleProjectChange` and `handleIssueTypeChange` functions.

```js
let fields = {
    project: {key: projectKey},
    summary: this.state.fields.summary,
    description: this.state.fields.description,
} as CreateIssueFields;
```

@maisnamrajusingh Please let me know if this makes sense and if it fixes the issue

Reply:
@mickmister I have updated the logic and it's working fine. However the user selector does seem to have a problem of it's own.  I know that it's because of the types not matching in the user selector but I guess as you have mentioned, that this will have effect on other options. I'll add it as an issue that maybe we can work on in the future. 
<img width="533" alt="Screenshot 2022-03-16 at 10 36 25 PM" src="https://user-images.githubusercontent.com/2786378/158629433-d6a0c864-c130-4dbe-8329-24af233faf72.png">

Reply:
@mickmister I have updated the logic and it's working fine. However the user selector does seem to have a problem of it's own.  I know that it's because of the types not matching in the user selector but I guess as you have mentioned, that this will have effect on other options. I'll add it as an issue that maybe we can work on in the future. <img width="533" alt="Screenshot 2022-03-16 at 10 36 25 PM" src="https://user-images.githubusercontent.com/2786378/158629433-d6a0c864-c130-4dbe-8329-24af233faf72.png">

Reply:
@mickmister this seems to be an existing issue that already existed and is not related to the modal crash. It's related to this line here 
https://github.com/mattermost/mattermost-plugin-jira/blob/f986c159ec37a6b1cf727840d8899f84a1aff612/webapp/src/components/data_selectors/jira_user_selector/jira_user_selector.tsx#L51

the return type should be 
```
  return {
                    value: user.displayName,
                    label,
                };
                ```
since the return type of the function is 
```
ReactSelectOption
```

Reply:
> however this doesn't have any effect on the way it works though.

Do you mean the feature still works correctly when we satisfy the prop type? Is it an error that `value: user` is being used, or does making it a string `user.displayName` break other logic?

It seems we need to integrate typescript into CI to catch these sort of errors.

Reply:
Ticket to add typescript to CI https://github.com/mattermost/mattermost-plugin-jira/issues/852

Reply:
Yeah the feature still works correctly after we satisfy user.displayName. I am worried that the component is used elsewhere and might break other components in case it's used eslewhere